src/main/java/uk/gov/digital/ho/pttg/application/HmrcCallWrapper.java

```diff
@@ -25,32 +27,34 @@
 
     private RestTemplate restTemplate;
     private TraversonFollower traversonFollower;
+    private ComponentTraceHeaderData componentTraceHeaderData;
 
-    public HmrcCallWrapper(@Qualifier("hmrcApiRestTemplate") RestTemplate restTemplate, TraversonFollower traversonFollower) {
+    public HmrcCallWrapper(@Qualifier("hmrcApiRestTemplate") RestTemplate restTemplate, TraversonFollower traversonFollower, ComponentTraceHeaderData componentTraceHeaderData) {
         this.restTemplate = restTemplate;
         this.traversonFollower = traversonFollower;
+        this.componentTraceHeaderData = componentTraceHeaderData;
     }
 
     @AbortIfBeyondMaxResponseDuration
     public <T> ResponseEntity<Resource<T>> exchange(URI uri, HttpMethod httpMethod, HttpEntity httpEntity, ParameterizedTypeReference<Resource<T>> reference) {
-        try {
-            return restTemplate.exchange(uri, httpMethod, httpEntity, reference);
-        } catch (HttpServerErrorException e) {
-            log.info("Received {} - {}", e.getStatusCode(), e.getStatusText());
-            throw e;
-        } catch (HttpClientErrorException e) {
-            throw handleClientErrorExceptions(e);
-        }
+        return handleExceptions(() -> restTemplate.exchange(uri, httpMethod, httpEntity, reference));
     }
 
     @AbortIfBeyondMaxResponseDuration
     <T> Resource<T> followTraverson(String link, String accessToken, ParameterizedTypeReference<Resource<T>> reference) {
+        return handleExceptions(() -> traversonFollower.followTraverson(link, accessToken, restTemplate, reference));
+    }
+
+    private <T> T handleExceptions(Supplier<T> hmrcRequest) {
```

Comment:
It's a bit of a shame you have to test this once per caller, when you are essentially testing the same thing.

Reply:
That is true. If I'm back in this area I may look to trim things. It almost feels like a new class.